libs/shared-atomic-design-components/src/lib/molecules/Paginate.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Pagination,
+  PaginationContent,
+  PaginationEllipsis,
+  PaginationItem,
+  PaginationLink,
+  PaginationNext,
+  PaginationPrevious,
+} from '@talent-connect/shared-shadcn-ui-components'
+import './Paginate.scss'
+
+interface PaginateProps {
+  totalPagesNumber: number
+  currentPageNumber: number
+  setCurrentPageNumber: (page: number) => void
+}
+
+type ActivePagesType = number | 'ellipsis'
+
+const MAX_VISIBLE_PAGES = 5
+
+const Paginate = ({
+  totalPagesNumber,
+  currentPageNumber,
+  setCurrentPageNumber,
+}: PaginateProps) => {
+  if (!totalPagesNumber) return null
+
+  const scrollToListTop = () => {
+    window.scrollTo({ top: 350, behavior: 'smooth' })
+  }
+
+  const handlePreviousPage = () => {
+    if (currentPageNumber > 1) {
+      setCurrentPageNumber(currentPageNumber - 1)
+    }
+    scrollToListTop()
+  }
+
+  const handlePageClick = (page: number) => {
+    setCurrentPageNumber(page)
+    scrollToListTop()
+  }
+
+  const handleNextPage = () => {
+    if (currentPageNumber < totalPagesNumber) {
+      setCurrentPageNumber(currentPageNumber + 1)
+    }
+    scrollToListTop()
+  }
+
+  const renderPages = () => {
+    const pageNumbers = [...Array(totalPagesNumber).keys()].map((i) => i + 1)
+    let activePages: ActivePagesType[] = []
+
+    // Determine pagination elements with proper ellipsis handling based on the current page's position
+    if (totalPagesNumber <= MAX_VISIBLE_PAGES) {
+      activePages = pageNumbers
+    } else if (currentPageNumber <= 3) {
+      activePages = [1, 2, 3, 'ellipsis', totalPagesNumber]
+    } else if (currentPageNumber >= totalPagesNumber - 2) {
+      activePages = [
+        1,
+        'ellipsis',
+        totalPagesNumber - 2,
+        totalPagesNumber - 1,
+        totalPagesNumber,
+      ]
+    } else {
+      activePages = [
+        1,
+        'ellipsis',
+        currentPageNumber,
+        'ellipsis',
+        totalPagesNumber,
+      ]
+    }
+
+    // Map activePages to rendered pagination items
+    const renderedPages = activePages.map((page, index) => {
+      if (page === 'ellipsis') {
+        return <PaginationEllipsis key={`ellipsis-${index}`} />
+      }
+      return (
+        <PaginationItem key={page}>
+          <PaginationLink
+            onClick={() => handlePageClick(page)}
+            isActive={currentPageNumber === page}
+          >
+            {page}
+          </PaginationLink>
+        </PaginationItem>
+      )
+    })
+
+    return renderedPages
```

Comment:
ChatGPT made this simplification suggestion:

```suggestion
    return activePages.map((page, index) => 
      page === 'ellipsis' ? (
        <PaginationEllipsis key={`ellipsis-${index}`} />
      ) : (
        <PaginationItem key={page}>
          <PaginationLink
            onClick={() => changePage(page)}
            isActive={currentPageNumber === page}
          >
            {page}
          </PaginationLink>
        </PaginationItem>
      )
    )
```

Reply:
What about this?
<img width="532" alt="image" src="https://github.com/user-attachments/assets/9180d05d-f427-49b0-bc3b-594acad7d04e">

Reply:
Looks good!